clang/lib/Driver/Driver.cpp

```diff
@@ -837,6 +849,16 @@ static llvm::Optional<StringRef> isIntelGPUTarget(StringRef Target) {
   return llvm::None;
 }
 
+static llvm::Optional<StringRef> isNvidiaGPUTarget(StringRef Target) {
+  // Handle target specifications that resemble 'nvidia_gpu_*' here. These are
+  // 'spir64_gen' based.
+  if (Target.startswith(NvidiaGPU)) {
+    return tools::SYCL::gen::resolveGenDevice(
+        Target.drop_front(sizeof(NvidiaGPU) - 1));
+  }
+  return llvm::None;
+}
```

Comment:
`isNvidiaGPUTarget` and `isIntelGPUTarget` are no longer used.

Reply:
@mdtoguchi this was still a wip to add support for amd and nvidia. that's fine now, supporting intel, amd and nvidia.